src/Core/Framework/Api/Controller/CacheController.php

```diff
@@ -80,8 +80,8 @@ public function __construct(
     public function info(): JsonResponse
     {
         return new JsonResponse([
-            'environment' => getenv('APP_ENV'),
-            'httpCache' => (bool) getenv('SHOPWARE_HTTP_CACHE_ENABLED'),
+            'environment' => $this->getParameter('kernel.environment'),
```

Comment:
@shyim This isn't the full service container. It is a service locator as Symfony does automatically if you use autowiring with the AbstractController. See below for corresponding links.

 I can but then I need to adjust the constructor which is definitely backwards incompatible. 

In any case, I switched to use the service locator which is how Symfony does it with autowiring for its AbstractController and the `getParameter` exists on the AbstractController. I can also inject the ParameterBag BUT Symfony provides the service locator and `getParameter` on the AbstractController that basically does this already.

Regarding service locator/subscriber: https://symfony.com/doc/current/service_container/service_subscribers_locators.html#defining-a-service-subscriber
https://github.com/symfony/symfony/blob/v5.4.6/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php#L80
https://github.com/symfony/symfony/blob/v5.4.6/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php#L64
https://github.com/symfony/symfony/blob/v5.4.6/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php#L57

Reply:
and one more link: https://symfony.com/doc/current/service_container/service_subscribers_locators.html#including-services

Reply:
but I can also inject the parameters themselves but due to service locators and the `getParameter` method on the AbstractController it is unnecessary.

Reply:
@shyim Btw. using https://github.com/symfony/symfony/blob/6.1/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php#L83 (you can extend it e.g. to add the event dispatcher)
in combination with switching to the service locator container `Psr\Container\ContainerInterface` during `setContainer´ would allow you to inject all necessary additional services by overriding that method and they do not need to be `public`.

that way instead of adding all new dependencies for the storefrontcontroller like this https://github.com/shopware/platform/blob/trunk/UPGRADE-6.4.md#storefront-controller-need-to-have-twig-injected-in-future-versions

you could adjust it manually by updating the service list, if the service definition is correctly done.

Reply:
as the controller is only constructed via the XML definition, adding new parameters to the constructor is no break 😉
it would only be a break, if the class is constructed with `new` somewhere

Reply:
@mitelg Not 100% true, one could overwrite the controller by extending it and if not using `parent` but completely overwriting it and its definition, it would then miss constructor arguments.
Though I think it isn't likely that this is ever done, if one intends to extend it, it would either be done as decorator or as decorator in combination with `parent` and then the issue doesn't exist.

In any case, I think using the service locator for parameter retrieval as intended by Symfony's AbstractController is still the best solution here. Espeically, as the parameter bag would only be needed for that one route. And injecting both parameters just makes the constructor list longer (same with parameter bag).

But if necessary I can change it. Doesn't mean the rest of the service configuration change shouldn't be done either as that is the correct way, in contrast to injecting the full service container. 
Should actually be done this way for all other controllers including storefront controllers and any necessary services should be added to the `getSubscribedServices` method return. That way, you would also be more secure regarding additional services you might need in e.g. Storefront controllers in the future, see other comments above

Reply:
We don't care about decoration of non interfaced/abstract class services. https://github.com/shopware/platform/blob/trunk/.bc-exclude.php#L22-L23

Reply:
@shyim A great. Makes sense. In any case, is the current version using the ServiceLocator fine then? After all you resolved the conversation.

Reply:
Is ok